src/services/slam/types.ts

```diff
@@ -0,0 +1,5 @@
+import tspb from 'google-protobuf/google/protobuf/timestamp_pb';
+import pb from '../../gen/service/slam/v1/slam_pb';
+
+export type Timestamp = tspb.Timestamp.AsObject;
```

Comment:
instead of exposing pb duration, we should convert this to a javascript date - i think the conversion for this is:

```ts
new Date(duration.getSeconds() * 1e3 + duration.getNanos() / 1e6)
```